Week 4/gemCollector/src/GemCollector.java

```diff
@@ -1,38 +1,42 @@
 import java.util.Stack;
 
+// The Gem Collector interpret the input
+// and calculates the maximum number of treasures that can be obtained on the given map,
+// moving from entrance to exit and prints the past route. 
+// The place that once passed cannot pass again.
+// The first line of the input should be in the form of 
+// (y-coordinate size)(space)(x-coordinate size).
+// Gem Collector recognizes 's' as the starting point, 'e' as the destination,'!' as an obstacle,
+// '*' as a treasure, and '.' as a road. Everything else is considered invalid input.
+// The map size should not exceed 15x15.
 public class GemCollector {
   // Save the information of the given map.
-  private PointInfo[][] pointInfoMap;
-  private int mapSizeY;
-  private int mapSizeX;
-  private Point startPoint;
-  private Point endPoint;
+  private boolean[][] gemMap;  // true if gem exists at the point.
+  private boolean[][] obstacleMap; // true if obstacle exists at the point.
+  private int mapSizeY = -1;
+  private int mapSizeX = -1;
+  private Point startPoint = new Point();
+  private Point endPoint = new Point();
   private Stack<PointWithMap> dfsStack = new Stack<>();
 
   // Save the answer.
-  int answerGem = -1;
-  boolean answerPath[][];
+  private int answerGem = -1;
+  private boolean[][] answerPath;
 
-  // Initialize the information of the given map.
-  public GemCollector(PointInfo[][] pointInfoMap, Point startPoint, Point endPoint) {
-    this.mapSizeY = pointInfoMap.length;
-    this.mapSizeX = pointInfoMap[0].length;
-    this.pointInfoMap = pointInfoMap;
-    this.startPoint = startPoint;
-    this.endPoint = endPoint;
-
-    this.answerPath = new boolean[mapSizeY][mapSizeX];
+  public GemCollector(String inputPath) throws Exception {
+    // Interpret the map in given file and set the variables.
+    InputManager inputManager = new InputManager(this);
+    inputManager.getInput(inputPath);
   }
 
-  // Find the maxim gem and the answer path.
+  // Collect the gems while traveling the path 
+  // and update maximum number of gems that can be collected and corresponding path.
   public int collect() {
-
-    // Push the start point to dfsStack.
-    // Initialize every element of the path map of the start point to false.
+    // Push the start point with path map to dfsStack.
     pushStartPoint();
```

Comment:
Why init in this function?? pushStartPoint()
This function works in two ways init and push start point. but name is pushstart point.
separate logic